src/timer/mod.rs

```diff
@@ -33,26 +43,58 @@ pub struct TimerResult {
     pub status: ExitStatus,
 }
 
+/// Discard the output of a child process.
+fn discard(output: ChildStdout) {
+    const CHUNK_SIZE: usize = 64 << 10;
+
+    #[cfg(target_os = "linux")]
+    {
+        if let Ok(file) = File::create("/dev/null") {
+            while let Ok(bytes) = splice(
+                output.as_raw_fd(),
+                None,
+                file.as_raw_fd(),
+                None,
+                CHUNK_SIZE,
+                SpliceFFlags::empty(),
+            ) {
+                if bytes == 0 {
+                    break;
+                }
+            }
+        }
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    {
+        let mut output = output;
+        let mut buf = [0; CHUNK_SIZE];
+        while let Ok(bytes) = output.read(&mut buf) {
+            if bytes == 0 {
+                break;
+            }
+        }
+    }
+}
+
 /// Execute the given command and return a timing summary
 pub fn execute_and_measure(mut command: Command) -> Result<TimerResult> {
     let wallclock_timer = WallClockTimer::start();
 
-    #[cfg(not(windows))]
-    let ((time_user, time_system), status) = {
-        let cpu_timer = self::unix_timer::CPUTimer::start();
-        let status = command.status()?;
-        (cpu_timer.stop(), status)
-    };
+    let mut child = command.spawn()?;
```

Comment:
The new implementation (`spawn` + `wait`) looks much cleaner than what I did before (`status`) - thanks! I'm not sure if I just didn't think of this way of doing it, or if there's actually a difference (on non-Windows systems) with this implementation? Starting the timer only after spawning shouldn't lead to measurable differences, I assume?

Reply:
There is a minor race window between `spawn()` and `start()` where the child could be running but not counted by the timer.  On a 2+ core machine it likely doesn't matter.  On a single-core machine you might get preempted for the child on `spawn()` and miss a full timeslice (a few ms) until `hyperfine` gets scheduled again.  Better safe than sorry:

```diff
diff --git a/src/timer/mod.rs b/src/timer/mod.rs
index 4a87c8c..4057318 100644
--- a/src/timer/mod.rs
+++ b/src/timer/mod.rs
@@ -81,17 +81,18 @@ fn discard(output: ChildStdout) {
 pub fn execute_and_measure(mut command: Command) -> Result<TimerResult> {
     let wallclock_timer = WallClockTimer::start();
 
-    let mut child = command.spawn()?;
-    let output = child.stdout.take();
-
     #[cfg(not(windows))]
     let cpu_timer = self::unix_timer::CPUTimer::start();
+
+    let mut child = command.spawn()?;
+
     #[cfg(windows)]
     let cpu_timer = self::windows_timer::CPUTimer::start_for_process(&child);
 
-    if let Some(output) = output {
+    if let Some(output) = child.stdout.take() {
         discard(output);
     }
+
     let status = child.wait()?;
 
     let (time_user, time_system) = cpu_timer.stop();
```